### DISCUSSION TITLE
efbc:TermCode lack in T02 ?

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#471] [Q&A] [answered]
Hi lab !
in T02 form : 
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/8b3a29e0-2601-4fe4-b7a8-e81fde7854bb)

CONDITIONS OTHER THAN REVENUE ALLOCATION (0001) and REWARDS AND PENALTIES became 

```                                                      
       <efac:ContractTerm>
		<efbc:TermCode listName="contract-detail">all-rev-tic</efbc:TermCode>
		<efbc:TermDescription languageID="ENG">CONDIZIONI CAPESTRO</efbc:TermDescription>
	</efac:ContractTerm>
	<efac:ContractTerm>
		<efbc:TermCode listName="rewards-penalties">rew-pen</efbc:TermCode>
		<efbc:TermDescription languageID="ENG">O PAGHI O TI DENUNCIO</efbc:TermDescription>
	</efac:ContractTerm>
```

Good, but what for Revenues allocation , that is OPP-032-Tender ? It remains 'alone' without efbc:TermCode and it gives , of course , xsd Error : 
```
<efac:ContractTerm>
    <efbc:TermPercent>400</efbc:TermPercent>
</efac:ContractTerm>
```
Can you tell me how to catch the 'third' <efbc:TermCode ?
Thanks and regards
Renato

**Comment by YvesJo** [comment]
There is no associated field defined and this element should be generated systematically as soon as a percentage for revenue allocation is specified. This may not be currently implemented in the editor demo.

**Reply by rengit62** [reply]
We do not use editor demo. We take 'inspiration' from editor demo especially in front-end, but we are building an new application based entirely on SDK.
What do you mean with "this element should be generated systematically as soon as a percentage for revenue allocation is specified" ? 
When the mandatory field OPP-032-Tender  is filled,  a second xml tag has to be generated ?
like <efbc:TermCode listName="contract-detail">all-rev-tic</efbc:TermCode> ?
It is a special rule for this notice or this field ? or for OPP-032-Tender ?


![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/63e4ce7c-cee9-48ff-8fee-a2fe936bec51)

**Comment by rengit62** [comment]
Hi Lab
I found an example notice Subtype T02 t02_ESP.xml and inside it the block : 
```
<efac:ContractTerm>
    <efbc:TermCode>all-rev-ic</efbc:TermCode>
    <efbc:TermIndicator>true</efbc:TermIndicator>
    <efbc:TermPercent>100</efbc:TermPercent>
</efac:ContractTerm>
```
I did not found anywhere, except in that notice, the string "all-rev-ic", I guess there no codelist in which is defined. 
Anyway as you wrote that the termCode should be generated systematically when TermPercent is specified ( it is mandatory, also .. ) , should I 'hardcode' all-rev-ic or anyValue ?
Bye
Renato

**Reply by YvesJo** [reply, marked as answer]
Hi,
The T02 mark-up example is incorrect and needs to be updated, "expected value should be "all-rev-tic" from the contract-detail codelist. Also efbc:TermIndicator has been inaccurately added to all terms. This will get fixed in a next release.
